erl/erl.go

```diff
@@ -0,0 +1,257 @@
+	// past 10 seconds.
+	RateWindow10s = fastly.RateWindow10s
+
+	// RateWindow60s indicates the rate of events per second over the
+	// past 60 seconds.
+	RateWindow60s = fastly.RateWindow60s
+)
+
+var (
+	// CounterDuration10s indicates the estimated number of events in
+	// the most recent 10 second bucket.
+	CounterDuration10s = fastly.CounterDuration10s
+
+	// CounterDuration20s indicates the estimated number of events in
+	// the most recent two 10 second buckets.
+	CounterDuration20s = fastly.CounterDuration20s
+
+	// CounterDuration30s indicates the estimated number of events in
+	// the most recent three 10 second buckets.
+	CounterDuration30s = fastly.CounterDuration30s
+	// CounterDuration40s indicates the estimated number of events in
+	// the most recent four 10 second buckets.
+	CounterDuration40s = fastly.CounterDuration40s
+
+	// CounterDuration50s indicates the estimated number of events in
+	// the most recent five 10 second buckets.
+	CounterDuration50s = fastly.CounterDuration50s
+
+	// CounterDuration60s indicates the estimated number of events in
+	// the most recent six 10 second buckets.
+	CounterDuration60s = fastly.CounterDuration60s
+)
+
+type (
+	// RateWindow indicates the rate of events per second in the current
+	// POP over one of a few predefined time windows.  See
+	// [RateWindow1s], [RateWindow10s], and [RateWindow60s].
+	RateWindow = fastly.RateWindow
+
+	// CounterDuration indicates the estimated number of events in this
+	// duration in the current POP.  Counts are divided into 10 second
+	// buckets, and each bucket represents the estimated number of
+	// requests received up to and including that 10 second window.
+	//
+	// Buckets are not continuous.  For example, if the current time is
+	// 12:01:03, then the 10 second bucket represents events received
+	// between 12:01:00 and 12:01:10, not between 12:00:53 and 12:01:03.
+	// This means that, in each minute at the ten second mark (:00, :10,
+	// :20, etc.) the window represented by each bucket will shift.
+	//
+	// Estimated counts are not precise and should not be used as
+	// counters.
+	//
+	// See [CounterDuration10s], [CounterDuration20s],
+	// [CounterDuration30s], [CounterDuration40s], [CounterDuration50s],
+	// and [CounterDuration60s].
+	CounterDuration = fastly.CounterDuration
+)
+
+// RateCounter is a named counter that can be incremented and queried.
+type RateCounter struct {
+	name string
+}
+
+// OpenRateCounter opens a rate counter with the given name, creating it
+// if it doesn't already exist.  The rate counter name may be up to 64
+// characters long.  Entry names in this counter are also limited to 64
+// characters.
+func OpenRateCounter(name string) *RateCounter {
+	return &RateCounter{name: name}
+}
+
+// Increment increments the rate counter for this entry by the given
+// delta value.  The minimum value is 0 and the maximum is 1000.
+func (rc *RateCounter) Increment(entry string, delta uint32) error {
```

Comment:
Should we also check these values in Go? It is fairly common in Go to allow "incrementing" by a negative number as well.

Reply:
> that's why it's a `uint32` here

Types, how do they work? 😵‍💫